### PR TITLE
feat: /boardowa command improvements

### LINKED ISSUE #105: Add upper/lower bound for table availability
What if we e.g. want to get best options between 6pm and 10pm?

### LINKED ISSUE #106: Parametrize the number of returned tables in `/boardowa` command

### PR DESCRIPTION
**Pull request by Iipin**
* refactor the command, split the logic into more submodules
* replace `Result<_, Box<dyn Error>>` with `anyhow::Result<_>`
* add params: `count`, `before`, `after`
* improve output formatting

Resolves #105, resolves #106

Example invocation and output:
image

### CODE REVIEW

File: src/slash_commands/boardowa.rs [hunk 1]
```diff
@@ -142,105 +105,99 @@ impl SlashCommand for BoardowaCommand {
...
+    format!(
+        "Table {} | {} ppl. | {}-{}",
+        replace_leading_zeros(range.table_id.as_str()),
+        range.capacity,
+        format_time(opening.from.into(), range.start),
+        format_time(opening.from.into(), range.start + (range.len as usize))
+    )
 }
 
-fn rank_tables(tables: Vec<TableInfo>) -> Vec<AvailabilityRange> {
```
Comment (m4tx) [inline src/slash_commands/boardowa.rs:238]: What is the purpose of this function? Just remove leading zeros and replace them with spaces? Maybe just [trim_start_matches](https://doc.rust-lang.org/std/primitive.str.html#method.trim_start_matches) the original string and add n times space, where n is the difference between new and old string lengths?

Comment (Iipin) [reply src/slash_commands/boardowa.rs:238]: Woah, didn't notice this function. Love this language

Comment (Iipin) [reply src/slash_commands/boardowa.rs:238]: (the purpose of this function was użycie `Cow` bo bardzo mi się podoba)

**Review by m4tx — APPROVED** [review]

**Comment by m4tx** [comment]
:shipit:

**Pull request merged**